<?xml version="1.0"?>
<launch>
    <!-- MAVROS posix SITL environment launch script -->
    <!-- launches MAVROS, PX4 SITL, Gazebo environment, and spawns vehicle -->


    <arg name="mundo" default="pruebas"/>
    <arg name="carpeta_imagenes" default="$(arg mundo)/"/>
    <arg name="angulo_entrada" default="0"/>

    <!-- vehicle model and world -->
    <arg name="gui" default="false"/>
    <arg name="vehicle" default="iris_gimbal"/>
    <arg name="world" default="$(find offbnode)/worlds/$(arg mundo).world"/>
    <!-- MAVROS configs -->
    <arg name="gcs_url" default="udp://:18570@localhost:14550"/>

    <!-- Gazebo sim -->
    <arg name="verbose" default="false"/>
    <arg name="paused" default="false"/>

<!-- nodos -->
    <include file="$(find offbnode)/launch/mavros_posix_sitl.launch">
        <arg name="world" value="$(arg world)"/>
        <arg name="paused" default="$(arg paused)"/>
        <arg name="verbose" default="$(arg verbose)"/>
        <arg name="vehicle" value="$(arg vehicle)"/>
        <arg name="gcs_url" value="$(arg gcs_url)"/>
        <arg name="gui" default="$(arg gui)"/>
    </include>
    <!-- <node type="offboard_node.py" name="offboard_node" pkg="offbnode" output="screen" /> -->
    <!-- <node type="image_joke.py" name="image_joke_cal" pkg="offbnode" output="screen" /> -->
    <!-- <node type="procesado_plc_2.py" name="procesado_plc_2" pkg="offbnode" output="screen" /> -->
    <node type="procesado_plc.py" name="procesado_plc" pkg="offbnode" output="screen" />
    <node type="maestro.py" name="maestro" pkg="offbnode" output="screen" />
    <node ns="offbnode" type="stereo_image_proc" name="stereo_image_proc" pkg="stereo_image_proc"
            output="screen" respawn="true">
        <param name="speckle_size" value="400" />
    </node>
    <!-- <node type="rute_plan.py" name="rute_plan" pkg="offbnode" output="screen" /> -->

    <!-- <node type="capture_for_px4.py" name="capture_for_px4" pkg="offbnode" output="screen" args="$(env HOME)/Capture_px4/$(arg carpeta_imagenes)" />
    <node type="offb_node_rute_square" name="offb_node_rute_square" pkg="offbnode" output="screen" args="$(arg angulo_entrada)" /> -->
    <!-- <node type="tensor_flow_prueba.py" name="tensor_flow_prueba" pkg="offbnode" output="screen" />
    <node type="tensor_flow_prueba2.py" name="tensor_flow_prueba2" pkg="offbnode" output="screen" /> -->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find offbnode)/rviz/nav.rviz" />

</launch>
